plexus-compilers/plexus-compiler-aspectj/src/main/java/org/codehaus/plexus/compiler/ajc/AspectJCompiler.java

```diff
@@ -557,83 +564,172 @@ private void checkForAspectJRT( List<String> cp )
     }
 
     /**
-     * Set the source version in ajc compiler
+     * Set the source version in AspectJ compiler
      *
      * @param buildConfig
      * @param sourceVersion
      */
     private void setSourceVersion( AjBuildConfig buildConfig, String sourceVersion )
         throws CompilerException
     {
-        if ( "11".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK11;
-        }
-        else if ( "10".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK10;
-        }
-        else if ( "9".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK9;
-        }
-        else if ( "1.9".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK9;
-        }
-        else if ( "1.8".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK1_8;
-        }
-        else if ( "1.7".equals( sourceVersion ) )
+        buildConfig.getOptions().sourceLevel = versionStringToMajorMinor( sourceVersion );
+    }
+
+    /**
+     * Set the target version in AspectJ compiler
+     *
+     * @param buildConfig
+     * @param targetVersion
+     */
+    private void setTargetVersion( AjBuildConfig buildConfig, String targetVersion )
+        throws CompilerException
+    {
+        buildConfig.getOptions().targetJDK = versionStringToMajorMinor( targetVersion );
+    }
+
+    private static final Map<String, Long> VERSION_MAP = new HashMap<>();
+
+    static
+    {
+        // Default in Maven Compiler 3.8.1
+        VERSION_MAP.put( "",     JDK1_6 );
```

Comment:
This looks like a weird assumption to me: a library that assumes it is being called by a specific version of a plugin. Should be removed.

Reply:
The comment only explains why I chose this default for an empty string in 2021. The version number is a hint about the origin of this choice. I can choose anything, but I wanted to document why I made the choice, so that in the future someone else can update the default, based on another up-to-date Maven Compiler version. Any default is OK, if it is documented. That is to yet be done, of course, because the Plexus components are lacking meaningful Maven sites.